controllers/gitrepository_controller.go

```diff
@@ -758,18 +760,23 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 	var gitReader git.RepositoryReader
 	var err error
 
-	switch obj.Spec.GitImplementation {
+	gitImplementation := obj.Spec.GitImplementation
+	if goGitOnly {
+		gitImplementation = sourcev1.GoGitImplementation
+	}
```

Comment:
Seems redundant to figure out the git implementation here again when something similar is done in `reconcileSource` already.
It may be better to pass the git implementation to `gitCheckout()` instead of `goGitOnly` flag.